### PR TITLE
Dockerize project's development setup

### PR DESCRIPTION
**Pull request by igordavydsson**
Refs #266

### CODE REVIEW

**Comment by Rikuoja** [comment]
Asking @jussiarpalahti for comments, as he's been doing containerization of our projects lately :)

**Comment by Rikuoja** [comment]
And @vikoivun will comment as our infrastructure and documentation person :P

File: local_settings.py.template [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    'default': {
+        'ENGINE': 'django.contrib.gis.db.backends.postgis',
+        'NAME': 'linkedevents',
+        'USER': 'linkedevents',
+        'PASSWORD': 'linkedevents',
+        'HOST': 'linkedevents-db'
+    }
+}
+
+LANGUAGES = (
```
Comment (Rikuoja) [inline local_settings.py.template]: I wouldn't have a template with different languages than the repo default, I guess this means there will be extra languages in linkedevents migrations that are not used by the code? Don't know if extra unused fields in the db will cause any trouble, though.

Comment (igordavydsson) [reply local_settings.py.template]: I could copypaste settings from other languages, maybe it would be ok for now. :D

**Comment by jussiarpalahti** [comment]
@igordavydsson Have you done Travis integration with docker compose before?
We have github -> Travis -> Docker Hub pipeline in the works. I'd would be nice to incorporate these changes and not do two separate build configs (the other one for dev).

File: docker/postgres/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM postgres:10.1
```
Comment (jussiarpalahti) [inline docker/postgres/Dockerfile]: We are currently stuck on Ubuntu 16.04 Postgres 9.5 so containers should reflect that.

File: docker/django/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+RUN apt-get update \
+    && apt-get install -y gdal-bin python-gdal python3-gdal
+
+# Let's install git packages from requirements.txt here
+RUN mkdir -p /src
+
+COPY requirements.txt /src/
+
+RUN pip install --upgrade pip \
+    && pip install --upgrade typing \
```
Comment (vikoivun) [inline docker/django/Dockerfile]: Why add typing? Especially through dockerfile?
Python 3.5 (and thus 3.6) will bring it along also…

File: docker/django/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,22 @@
+FROM python:3.4
+
+ENV PYTHONUNBUFFERED 0
+
+RUN apt-get update \
+    && apt-get install -y gdal-bin python-gdal python3-gdal
+
+# Let's install git packages from requirements.txt here
+RUN mkdir -p /src
```
Comment (vikoivun) [inline docker/django/Dockerfile]: Instead of this, you can shorten the dockerfile a bit and use /usr/src

File: docker/django/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+COPY requirements.txt /src/
+
+RUN pip install --upgrade pip \
+    && pip install --upgrade typing \
+    && pip install pip-tools \
+    && pip install -r /src/requirements.txt
```
Comment (vikoivun) [inline docker/django/Dockerfile]: Use `pip install --src /usr/src -r requirements.txt` to install in /usr/src. Nifty! Saves three lines!

File: docker/postgres/docker-entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+set -e
+
+psql -v ON_ERROR_STOP=1 --username "$POSTGRES_USER" --dbname "linkedevents" <<-EOSQL
+	CREATE EXTENSION postgis;
```
Comment (vikoivun) [inline docker/postgres/docker-entrypoint.sh]: CREATE EXTENSION postgis IF NOT EXISTS (and the same for hstore)
is a bit more robust.

File: docker/postgres/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+
+RUN apt-get update \
+    && apt-get install -y postgis
+
+RUN localedef -i fi_FI -c -f UTF-8 -A /usr/share/locale/locale.alias fi_FI.UTF-8
```
[the remark this answers is not in the document]
Comment (vikoivun) [reply docker/postgres/Dockerfile]: Ah. I see, whether it is or not is irrelevant. The environment set for the container is used to create the initial database. I guess that is sensible.

[the remark this answers is not in the document]
Comment (igordavydsson) [reply docker/postgres/Dockerfile]: yeah, making sure that the database uses the right locale

File: docker/postgres/Dockerfile (hunk 2 above)
Comment (igordavydsson) [reply docker/postgres/Dockerfile]: alright! switchin' to postgres 9.5 image, which is based on `debian-slim`, I guess that should be okay

File: docker/django/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,16 @@
+FROM python:3.4
```
[the remark this answers is not in the document]
Comment (igordavydsson) [reply docker/django/Dockerfile]: ah, sure, will do

File: docker/django/Dockerfile (hunk 4 above)
Comment (igordavydsson) [reply docker/django/Dockerfile]: oh, yeah, good idea! will use that 🙂

File: docker/django/Dockerfile (hunk 5 above)
Comment (igordavydsson) [reply docker/django/Dockerfile]: nifty indeed!

File: docker/postgres/docker-entrypoint.sh (hunk 6 above)
Comment (igordavydsson) [reply docker/postgres/docker-entrypoint.sh]: 👍

**Comment by igordavydsson** [comment]
@jussiarpalahti @vikoivun thanks for the comments! tried to address all of them.

@jussiarpalahti I haven't done Travis integration before, but I've seen a few tutorials on how to do that. Could be done for this project as well, but would take some time and I don't have it right now during this sprint. Maybe later there would be a chance to implement it. Anyway, I think such integration could be later based on this docker setup with some tweaks and adjustments, of course.

**Pull request merged**